airbyte-json-validation/src/main/java/io/airbyte/validation/json/JsonSchemaValidator.java

```diff
@@ -63,6 +66,51 @@ protected JsonSchemaValidator(URI baseUri) {
     this.baseUri = baseUri;
   }
 
+  /**
+   * Create and cache a schema validator for a particular schema. This validator is used when
+   * {@link #testInitializedSchema(String, JsonNode)} and
+   * {@link #ensureInitializedSchema(String, JsonNode)} is called.
+   */
+  public void initializeSchemaValidator(String schemaName, JsonNode schemaJson) {
+    schemaToValidators.put(schemaName, getSchemaValidator(schemaJson));
+  }
+
+  public boolean testInitializedSchema(String schemaName, JsonNode objectJson) {
+    final var validate = schemaToValidators.get(schemaName).validate(objectJson);
+    return validate.isEmpty();
+  }
+
+  public void ensureInitializedSchema(final String schemaName, final JsonNode objectNode) throws JsonValidationException {
+    final Set<ValidationMessage> validationMessages = schemaToValidators.get(schemaName).validate(objectNode);
+    if (validationMessages.isEmpty()) {
+      return;
+    }
+
+    throw new JsonValidationException(
+        String.format(
+            "json schema validation failed when comparing the data to the json schema. \nErrors: %s \nSchema: \n%s", Strings.join(validationMessages,
```

Comment:
any reason why we use our `Strings.join` instead of java's [String.join](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/String.html#join(java.lang.CharSequence,java.lang.Iterable))

Reply:
I see that our implementation accepts an `Iterable<?>` instead of an `Iterable<? extends CharSequence>`.  Unrelated to this PR but we should update our `Strings.join` to match the parameter ordering of `String.join` (have the delimiter be the first parameter).

Reply:
Not 100%. This predates even me. 

@michel-tricot git says this is you, can you recall why we don't use Strings.join?

Reply:
@colesnodgrass I was just following the existing pattern in the class. I'll move to String.join. Let's tackle Strings.join refactor in a follow up PR to keep this manageable. Lmk if you feel strongly.

Reply:
On closer look, I think I know why - `Iterable<?>` allows us to pass in a collection, while `Iterable<? extends CharSequence>` basically means the parameter is limited to Strings.

In any case, I'll follow up in a separate PR.

Reply:
No idea :)